libcontainer/rootfs_linux.go

```diff
@@ -25,14 +25,28 @@ import (
 
 const defaultMountFlags = syscall.MS_NOEXEC | syscall.MS_NOSUID | syscall.MS_NODEV
 
+// setupDev returns true if /dev needs to be set up.
+func needsSetupDev(config *configs.Config) (bool, error) {
+	setupDev := true
+	for _, m := range config.Mounts {
+		if m.Device == "bind" && m.Destination == "/dev" {
```

Comment:
What we did in docker before, when working directly with libcontainer, is check if the source is `/dev` and the destination has a `/dev/` prefix.

This is implementation is probably ok. If someone is mounting their own `/dev` I would expect it to be pre-configured and not net need the extra check for if it's the host's `/dev`.

(that said, I've not tested yet, just reading through the code)